src/transaction.rs

```diff
@@ -298,58 +337,65 @@ impl Transaction {
 
     pub const fn metadata(&self) -> Option<&Metadata> {
         match self {
-            Self::Script { metadata, .. } => metadata.as_ref(),
-            Self::Create { metadata, .. } => metadata.as_ref(),
+            Self::Script { metadata, .. }
+            | Self::Create { metadata, .. }
+            | Self::Mint { metadata, .. } => metadata.as_ref(),
         }
     }
 
-    pub const fn salt(&self) -> Option<&Salt> {
+    pub const fn salt(&self) -> Result<&Salt, Error> {
```

Comment:
We shouldn't use Result for a "maybe" result, these should be `Option` like it was before.